go/signedexchange/verifier.go

```diff
@@ -115,16 +116,47 @@ func (e *Exchange) Verify(verificationTime time.Time, certFetcher CertFetcher, l
 			l.Printf("validity-url (%s) is not same-origin with request URL (%v)", signature.ValidityUrl, e.RequestURI)
 			continue
 		}
+
 		// Step 2: "Use Section 3.5 to determine the signature's validity for
-		// requestUrl, headers, and payload, getting certificate-chain back. If
-		// this returned "invalid" or didn't return a certificate chain, return
-		// "invalid"."
+		//         requestUrl, headers, and payload, getting certificate-chain
+		//         back. If this returned "invalid" or didn't return a
+		//         certificate chain, return "invalid"."
 		_, err = verifySignature(e, verificationTime, certFetcher, signature)
 		if err != nil {
 			l.Printf("Verification of sinature %q failed: %v", signature.Label, err)
 			continue
 		}
-		// TODO: Implement Step 3-7.
+
+		// Step 3: "Let exchange be the exchange metadata and headers parsed out
+		//         of headers."
+		// e contains the exchange metadata and headers.
```

Comment:
L132 seems broken?

Reply:
Not actually, 'e' refers to a local variable.
Added `` ` `` around `e` hoping to make it easier to understand.